### DISCUSSION TITLE
Always include CSS in revealjs slides

### DISCUSSION THREAD

**Discussion by sagikazarmark** [quarto-dev/quarto-cli#4182] [Feature Requests]
I want to include a certain piece of CSS in all of my slides (regardless of the theme), but still want to be able to use custom CSS on a per slide basis if I want to.

Currently if I set a default value for `theme` with a custom css file, it gets overridden as soon as I pick another theme in a slide (which is kind of expected).

I'd like to be able to include a CSS file in addition to the theme.

A good enough workaround for now is using `include-in-header`, but that can still be overridden.

**Comment by cderv** [comment]
> Currently if I set a default value for theme with a custom css file, it gets overridden as soon as I pick another theme in a slide (which is kind of expected).

Can you share a reproducible example of what you are currently doing ? That would help us understand correctly all the context. In particular, how to you assign another theme to a slide ? Thank you

**Reply by sagikazarmark** [reply]
Sure!

Here is what I have in my `_quarto.yml` file:

```yaml
format:
  revealjs: 
    theme: [default, custom.scss]
```

Then in any slide, if I want to use another theme:

```markdown
---
format:
  revealjs: 
    theme: sky
---
```

I want to be able to include my `custom.scss` in all of my slides without having to remember to include it when I don't use the default value.

What I'm currently doing is I have `include-in-header` defined in my `_quarto.yml` to include custom CSS. I don't use `include-in-header` at the moment anywhere in my slides which is why I can rely on the default.

**Reply by cderv** [reply]
Oh I see. This is indeed not possible as `theme` in main YAML will be replaced by the one in local file, and currently custom theme needs to be passed at same layer as `theme` list. 

It would require another mechanism to allow that 🤔 

I'll open an issue to think about it further

**Reply by sagikazarmark** [reply]
Appreciate it, thank you. It might probably make sense to think about this in a broader context.

For example: I'm pretty sure that the same would be useful for websites as well (not sure how the config works currently, but I'M guessing it's similar to reveal).

I actually like the fact that there is an overriding mechanism, so I definitely agree that this should be some other, global configuration.

**Comment by mcanouil** [comment]
Closing in favour of https://github.com/quarto-dev/quarto-cli/issues/4193
